pinot-spi/src/main/java/org/apache/pinot/spi/config/table/SegmentsValidationAndRetentionConfig.java

```diff
@@ -124,6 +124,9 @@ public void setSegmentPushType(String segmentPushType) {
     _segmentPushType = segmentPushType;
   }
 
+  /**
```

Comment:
Don't deprecate them. The one in table config is just a helper method. They are also needed for json ser/de

Reply:
removed the deprecation annotation but kept the comment to guide ppl to use `TableConfig#getReplication()`